### PR TITLE
Adding SqlConnectionStringBuilder.Enlist to public method contract

### PR DESCRIPTION
**Pull request by geleems**
I added `SqlConnectionStringBuilder.Enlist` to public method contract, and also add more unit tests for `SqlConnectionStringBuilder.Enlist`

### CODE REVIEW

**Comment by corivera** [comment]
Can you also remove the ambient transaction not supported error message from the strings.resx?

File: src/System.Data.SqlClient/src/Resources/Strings.resx [hunk 1]
```diff
@@ -1180,7 +1180,4 @@
   <data name="MDF_UnableToBuildCollection" xml:space="preserve">
     <value>Unable to build schema collection '{0}';</value>
   </data>
-  <data name="AmbientTransactionsNotSupported" xml:space="preserve">
```
Comment (corivera) [inline src/System.Data.SqlClient/src/Resources/Strings.resx]: You'll also need to remove the exceptions that reference this. Build the project, and they'll show up in the build errors.

Comment (geleems) [inline src/System.Data.SqlClient/src/Resources/Strings.resx]: done.

**Comment by danmosemsft** [comment]
@gleems @corivera BTW, missing members vs. desktop are listed here, in case you want to see what else is missing from desktop: 

https://github.com/dotnet/corefx/blob/master/src/shims/ApiCompatBaseline.netcoreapp.netfx461.txt

I'll regenerate that file to remove these two.

**Pull request merged**